File: scip-snapshots/expected/java/common/scip-snapshots/cases/java/common/src/main/java/minimized/Enums.java
  package minimized;
  
  import java.util.Arrays;
//       ^^^^ reference scip-java maven . . java/
//            ^^^^ reference scip-java maven . . java/util/
//                 ^^^^^^ reference scip-java maven jdk 11 java/util/Arrays#
  import java.util.stream.Collectors;
//       ^^^^ reference scip-java maven . . java/
//            ^^^^ reference scip-java maven . . java/util/
//                 ^^^^^^ reference scip-java maven . . java/util/stream/
//                        ^^^^^^^^^^ reference scip-java maven jdk 11 java/util/stream/Collectors#
  
//⌄ enclosing_range_start scip-java maven . . minimized/Enums#
  enum Enums {
//     ^^^^^ definition scip-java maven . . minimized/Enums#
//           kind Enum
//           display_name Enums
//           signature_documentation
//           > enum Enums
//           relationship scip-java maven jdk 11 java/io/Serializable# implementation
//           relationship scip-java maven jdk 11 java/lang/Comparable# implementation
//           relationship scip-java maven jdk 11 java/lang/Enum# implementation
//           relationship scip-java maven jdk 11 java/lang/constant/Constable# implementation
//  ⌄ enclosing_range_start scip-java maven . . minimized/Enums#A.
    A("A", 420),
//  ^ definition scip-java maven . . minimized/Enums#A.
//    display_name A("A", 420)
//    signature_documentation
//    > Enums.A("A", 420) /* ordinal 0 */
//  ^ reference scip-java maven . . minimized/Enums#`<init>`().
//            ⌃ enclosing_range_end scip-java maven . . minimized/Enums#A.
//  ⌄ enclosing_range_start scip-java maven . . minimized/Enums#B.
    B("B", 1),
//  ^ definition scip-java maven . . minimized/Enums#B.
//    display_name B("B", 1)
//    signature_documentation
//    > Enums.B("B", 1) /* ordinal 1 */
//  ^ reference scip-java maven . . minimized/Enums#`<init>`().
//          ⌃ enclosing_range_end scip-java maven . . minimized/Enums#B.
//  ⌄ enclosing_range_start scip-java maven . . minimized/Enums#C.
    C("C", 5);
//  ^ definition scip-java maven . . minimized/Enums#C.
//    display_name C("C", 5)
//    signature_documentation
//    > Enums.C("C", 5) /* ordinal 2 */
//  ^ reference scip-java maven . . minimized/Enums#`<init>`().
//          ⌃ enclosing_range_end scip-java maven . . minimized/Enums#C.
//  ⌄ enclosing_range_start scip-java maven . . minimized/Enums#value.
    public String value;
//         ^^^^^^ reference scip-java maven jdk 11 java/lang/String#
//                ^^^^^ definition scip-java maven . . minimized/Enums#value.
//                      kind Field
//                      display_name value
//                      signature_documentation
//                      > public String value
//                     ⌃ enclosing_range_end scip-java maven . . minimized/Enums#value.
  
//  ⌄ enclosing_range_start scip-java maven . . minimized/Enums#`<init>`().
//        ⌄ enclosing_range_start local 0
//                      ⌄ enclosing_range_start local 1
    Enums(String value, int a) {
//  ^^^^^ definition scip-java maven . . minimized/Enums#`<init>`().
//        kind Constructor
//        display_name <init>
//        signature_documentation
//        > private Enums(String value, int a)
//        ^^^^^^ reference scip-java maven jdk 11 java/lang/String#
//               ^^^^^ definition local 0
//                     display_name value
//                     signature_documentation
//                     > String value
//                          ^ definition local 1
//                            display_name a
//                            signature_documentation
//                            > int a
//                   ⌃ enclosing_range_end local 0
//                          ⌃ enclosing_range_end local 1
      this.value = value;
//         ^^^^^ reference scip-java maven . . minimized/Enums#value.
//                 ^^^^^ reference local 0
    }
//  ⌃ enclosing_range_end scip-java maven . . minimized/Enums#`<init>`().
  
//  ⌄ enclosing_range_start scip-java maven . . minimized/Enums#app().
    public static String app() {
//                ^^^^^^ reference scip-java maven jdk 11 java/lang/String#
//                       ^^^ definition scip-java maven . . minimized/Enums#app().
//                           kind StaticMethod
//                           display_name app
//                           signature_documentation
//                           > public static String app()
//    ⌄ enclosing_range_start local 2
//                                             ⌄ enclosing_range_start local 3
      String all = Arrays.stream(values()).map(e -> e.value).map(Enums::valueOf).collect(Collectors.toList()).toString();
//    ^^^^^^ reference scip-java maven jdk 11 java/lang/String#
//           ^^^ definition local 2
//               kind Variable
//               display_name all
//               signature_documentation
//               > String all
//                 ^^^^^^ reference scip-java maven jdk 11 java/util/Arrays#
//                        ^^^^^^ reference scip-java maven jdk 11 java/util/Arrays#stream().
//                               ^^^^^^ reference scip-java maven . . minimized/Enums#values().
//                                         ^^^ reference scip-java maven jdk 11 java/util/stream/Stream#map().
//                                             ^ definition local 3
//                                               display_name e
//                                               signature_documentation
//                                               > Enums e
//                                                  ^ reference local 3
//                                                    ^^^^^ reference scip-java maven . . minimized/Enums#value.
//                                                           ^^^ reference scip-java maven jdk 11 java/util/stream/Stream#map().
//                                                               ^^^^^ reference scip-java maven . . minimized/Enums#
//                                                                      ^^^^^^^ reference scip-java maven . . minimized/Enums#valueOf().
//                                                                               ^^^^^^^ reference scip-java maven jdk 11 java/util/stream/Stream#collect(+1).
//                                                                                       ^^^^^^^^^^ reference scip-java maven jdk 11 java/util/stream/Collectors#
//                                                                                                  ^^^^^^ reference scip-java maven jdk 11 java/util/stream/Collectors#toList().
//                                                                                                            ^^^^^^^^ reference scip-java maven jdk 11 java/lang/Object#toString().
//                                             ⌃ enclosing_range_end local 3
//                                                                                                                      ⌃ enclosing_range_end local 2
      return all + A.value + B.value + C.value;
//           ^^^ reference local 2
//                 ^ reference scip-java maven . . minimized/Enums#A.
//                   ^^^^^ reference scip-java maven . . minimized/Enums#value.
//                           ^ reference scip-java maven . . minimized/Enums#B.
//                             ^^^^^ reference scip-java maven . . minimized/Enums#value.
//                                     ^ reference scip-java maven . . minimized/Enums#C.
//                                       ^^^^^ reference scip-java maven . . minimized/Enums#value.
    }
//  ⌃ enclosing_range_end scip-java maven . . minimized/Enums#app().
  }
//⌃ enclosing_range_end scip-java maven . . minimized/Enums#
  

File: scip-snapshots/expected/kotlin/common/scip-snapshots/cases/kotlin/common/src/main/kotlin/snapshots/Enums.kt
  package snapshots
//        ^^^^^^^^^ reference scip-java maven . . snapshots/
  
//⌄ enclosing_range_start scip-java maven . . snapshots/Suit#
//⌄ enclosing_range_start scip-java maven . . snapshots/Suit#values().
//⌄ enclosing_range_start scip-java maven . . snapshots/Suit#valueOf().
//⌄ enclosing_range_start scip-java maven . . snapshots/Suit#valueOf().(value)
//⌄ enclosing_range_start scip-java maven . . snapshots/Suit#entries.
//⌄ enclosing_range_start scip-java maven . . snapshots/Suit#getEntries().
//           ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#`<init>`().
//                ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#`<init>`().(symbol)
//                ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#symbol.
//                ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#getSymbol().
  enum class Suit(val symbol: Char) {
//           ^^^^ definition scip-java maven . . snapshots/Suit#
//                display_name Suit
//                signature_documentation
//                > public final enum class Suit : Enum<Suit>
//                relationship scip-java maven . . kotlin/Enum# implementation
//           ^^^^ definition scip-java maven . . snapshots/Suit#`<init>`().
//                display_name Suit
//                signature_documentation
//                > private constructor(symbol: Char): Suit
//           ^^^^ definition scip-java maven . . snapshots/Suit#values().
//                display_name values
//                signature_documentation
//                > public final static fun values(): Array<Suit>
//           ^^^^ definition scip-java maven . . snapshots/Suit#valueOf().
//                display_name valueOf
//                signature_documentation
//                > public final static fun valueOf(value: String): Suit
//           ^^^^ definition scip-java maven . . snapshots/Suit#valueOf().(value)
//                display_name value
//                signature_documentation
//                > value: String
//           ^^^^ definition scip-java maven . . snapshots/Suit#entries.
//                display_name entries
//                signature_documentation
//                > public final static val entries: EnumEntries<Suit>
//           ^^^^ definition scip-java maven . . snapshots/Suit#getEntries().
//                display_name entries
//                signature_documentation
//                > public get(): EnumEntries<Suit>
//                    ^^^^^^ definition scip-java maven . . snapshots/Suit#`<init>`().(symbol)
//                           display_name symbol
//                           signature_documentation
//                           > symbol: Char
//                    ^^^^^^ definition scip-java maven . . snapshots/Suit#symbol.
//                           display_name symbol
//                           signature_documentation
//                           > public final val symbol: Char
//                    ^^^^^^ reference scip-java maven . . snapshots/Suit#`<init>`().(symbol)
//                    ^^^^^^ definition scip-java maven . . snapshots/Suit#getSymbol().
//                           display_name symbol
//                           signature_documentation
//                           > public get(): Char
//                            ^^^^ reference scip-java maven . . kotlin/Char#
//                               ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#`<init>`().(symbol)
//                               ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#symbol.
//                               ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#getSymbol().
//                                ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#`<init>`().
//    ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#HEARTS.
      HEARTS('h'),
//    ^^^^^^ definition scip-java maven . . snapshots/Suit#HEARTS.
//           display_name HEARTS
//           signature_documentation
//           > public final val HEARTS: Suit
//               ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#HEARTS.
//    ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#SPADES.
      SPADES('s');
//    ^^^^^^ definition scip-java maven . . snapshots/Suit#SPADES.
//           display_name SPADES
//           signature_documentation
//           > public final val SPADES: Suit
//               ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#SPADES.
  
//    ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#isRed().
      fun isRed(): Boolean = symbol == 'h'
//        ^^^^^ definition scip-java maven . . snapshots/Suit#isRed().
//              display_name isRed
//              signature_documentation
//              > public final fun isRed(): Boolean
//                 ^^^^^^^ reference scip-java maven . . kotlin/Boolean#
//                           ^^^^^^ reference scip-java maven . . snapshots/Suit#symbol.
//                           ^^^^^^ reference scip-java maven . . snapshots/Suit#getSymbol().
//                                  ^^ reference scip-java maven . . kotlin/Char#equals().
//                                       ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#isRed().
  
//    ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#Companion#
//    ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#Companion#`<init>`().
      companion object {
//    ^^^^^^^^^^^^^^^^^^ definition scip-java maven . . snapshots/Suit#Companion#
//                       display_name Companion
//                       signature_documentation
//                       > public final companion object Companion : Any
//    ^^^^^^^^^^^^^^^^^^ definition scip-java maven . . snapshots/Suit#Companion#`<init>`().
//                       display_name Companion
//                       signature_documentation
//                       > private constructor(): Suit.Companion
//        ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#Companion#fromSymbol().
//                       ⌄ enclosing_range_start scip-java maven . . snapshots/Suit#Companion#fromSymbol().(symbol)
//                                                           ⌄ enclosing_range_start local 0
          fun fromSymbol(symbol: Char): Suit? = entries.find { it.symbol == symbol }
//            ^^^^^^^^^^ definition scip-java maven . . snapshots/Suit#Companion#fromSymbol().
//                       display_name fromSymbol
//                       signature_documentation
//                       > public final fun fromSymbol(symbol: Char): Suit?
//                       ^^^^^^ definition scip-java maven . . snapshots/Suit#Companion#fromSymbol().(symbol)
//                              display_name symbol
//                              signature_documentation
//                              > symbol: Char
//                               ^^^^ reference scip-java maven . . kotlin/Char#
//                                      ^^^^ reference scip-java maven . . snapshots/Suit#
//                                              ^^^^^^^ reference scip-java maven . . snapshots/Suit#entries.
//                                              ^^^^^^^ reference scip-java maven . . snapshots/Suit#getEntries().
//                                                      ^^^^ reference scip-java maven . . kotlin/collections/find(+9).
//                                                           ^^^^^^^^^^^^^^^^^^^^^^^ definition local 0
//                                                                                   display_name it
//                                                                                   signature_documentation
//                                                                                   > it: Suit
//                                                             ^^ reference local 0
//                                                                ^^^^^^ reference scip-java maven . . snapshots/Suit#symbol.
//                                                                ^^^^^^ reference scip-java maven . . snapshots/Suit#getSymbol().
//                                                                       ^^ reference scip-java maven . . kotlin/Char#equals().
//                                                                          ^^^^^^ reference scip-java maven . . snapshots/Suit#Companion#fromSymbol().(symbol)
//                                  ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#Companion#fromSymbol().(symbol)
//                                                                                 ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#Companion#fromSymbol().
//                                                                                 ⌃ enclosing_range_end local 0
      }
//    ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#Companion#
//    ⌃ enclosing_range_end scip-java maven . . snapshots/Suit#Companion#`<init>`().
  }
//⌃ enclosing_range_end scip-java maven . . snapshots/Suit#
//⌃ enclosing_range_end scip-java maven . . snapshots/Suit#values().
//⌃ enclosing_range_end scip-java maven . . snapshots/Suit#valueOf().
//⌃ enclosing_range_end scip-java maven . . snapshots/Suit#valueOf().(value)
//⌃ enclosing_range_end scip-java maven . . snapshots/Suit#entries.
//⌃ enclosing_range_end scip-java maven . . snapshots/Suit#getEntries().
  
//⌄ enclosing_range_start scip-java maven . . snapshots/describe().
//             ⌄ enclosing_range_start scip-java maven . . snapshots/describe().(suit)
  fun describe(suit: Suit): String =
//    ^^^^^^^^ definition scip-java maven . . snapshots/describe().
//             display_name describe
//             signature_documentation
//             > public final fun describe(suit: Suit): String
//             ^^^^ definition scip-java maven . . snapshots/describe().(suit)
//                  display_name suit
//                  signature_documentation
//                  > suit: Suit
//                   ^^^^ reference scip-java maven . . snapshots/Suit#
//                          ^^^^^^ reference scip-java maven . . kotlin/String#
//                      ⌃ enclosing_range_end scip-java maven . . snapshots/describe().(suit)
//          ⌄ enclosing_range_start local 1
      when (suit) {
//          ^^^^ definition local 1
//               display_name <when-subject>
//               signature_documentation
//               > local val <when-subject>: Suit
//          ^^^^ reference scip-java maven . . snapshots/describe().(suit)
//             ⌃ enclosing_range_end local 1
          Suit.HEARTS -> "red"
//        ^^^^ reference scip-java maven . . snapshots/Suit#
//             ^^^^^^ reference scip-java maven . . snapshots/Suit#HEARTS.
          Suit.SPADES -> "black"
//        ^^^^ reference scip-java maven . . snapshots/Suit#
//             ^^^^^^ reference scip-java maven . . snapshots/Suit#SPADES.
      }
//    ⌃ enclosing_range_end scip-java maven . . snapshots/describe().
  
